src/main/java/io/sqreen/sasdk/backend/IngestionHttpClient.java

```diff
@@ -47,7 +44,7 @@
          * {@link AuthenticationException}, {@link BadHttpStatusException} and
          * {@link InvalidPayloadException}.
          */
-        void reportBatch(Collection<?> signalsAndTraces, Multimap<String, String> headers) throws IOException;
+        void reportBatch(Collection<Signal> signalsAndTraces, Multimap<String, String> headers) throws IOException;
```

Comment:
We need the ability to submit arbitrary `Object`s, as we use this functionality in the agent. If you want to make the interfaces take only `Signal` types, then it's better to create separate interfaces or to parameterize the current ones with the types of signals and traces they take (however, if you parameterize the current ones, you'll need to have separate implementation classes, as you cannot have `class A implements Foo<B>, Foo<C> {}`).